core/corehttp/gateway_handler.go

```diff
@@ -205,9 +228,17 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 				}
 			}
 
+			// strip /ipfs/$hash from backlink if IPNSHostnameOption touched the path.
+			// we could use originalUrlPath like we do with the redirect,
+			// but i don't wanna break the switch-case right above, not now.
+			if urlPath != originalUrlPath {
+				backLink = "/" + strings.Join(pathSplit[4:], "/")
```

Comment:
is that the only case where `urlPath != originalUrlPath`? because if not, we might panic the process (by accessing `pathSplit[4:]`. 

Maybe:
```go
if urlPath != originalUrlPath && pathSplit[1] == "ipns" {
  backLink = "/" + strings.Join(pathSplit[4:], "/")
}
```
also, shouldn't it be `pathSplit[3:]` (3) ?